### PR TITLE
Feature/default song cover

### PR DESCRIPTION
**Pull request by FroggyRocky**
I improved getCoverBase64 function, before when getCoverBase64 function failed to fetch a cover from spotify api, it wouldn't show the whole badge, now it checks if a cover fetched successfully and if a cover exists, if it doesn't, it shows a default Spotify logo. The PM solves the issue [#8](https://github.com/akellbl4/spotify-badge/issues/8)

### CODE REVIEW

**Comment by FroggyRocky** [comment]
@akellbl4

**Comment by akellbl4** [comment]
@FroggyRocky can package updates be avoided in the PR? If packages have to be updated it is better to file a separate PR.
Also, the project uses pnpm and not npm

File: components/Player.tsx [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import { h } from 'preact'
-
```
Comment (akellbl4) [inline components/Player.tsx]: please avoid unrelated changes

File: lib/spotify.ts [hunk 2]
```diff
@@ -80,10 +80,14 @@ async function getCurrentTrack(): Promise<null | TrackInfo> {
...
 
 async function getCoverBase64(url: string) {
+	try {
 	const res = await fetch(url)
 	const buff = await res.arrayBuffer()
-
 	return Buffer.from(buff).toString('base64')
+	} catch(e) {
+		console.error('Error fetching cover image:', e)
+		return null
```
Comment (akellbl4) [inline lib/spotify.ts]: we can return empty string and avoid fallback in `getNowPlaying`

File: package.json [hunk 3]
```diff
@@ -10,17 +10,19 @@
...
   "dependencies": {
+    "@types/react": "^18.3.12",
+    "@types/react-dom": "^18.3.1",
     "isomorphic-unfetch": "^3.1.0",
     "preact": "^10.8.1",
     "preact-render-to-string": "^5.2.0"
   },
   "devDependencies": {
-    "@types/node": "^18.0.0",
+    "@types/node": "^14.18.63",
```
Comment (akellbl4) [inline package.json]: the types are downgraded from 18 to 14

**Review by akellbl4 — CHANGES_REQUESTED** [review]

File: tsconfig.json [hunk 4]
```diff
@@ -1,14 +1,15 @@
 {
   "compilerOptions": {
     "baseUrl": ".",
-    "module": "NodeNext",
-    "moduleResolution": "NodeNext",
+    "module": "esnext",
```
Comment (akellbl4) [inline tsconfig.json]: The config have been recently updated to fix the problem with deployment to Vercel
https://github.com/akellbl4/spotify-badge/commit/f82cbcfb42bad06021244b11a957e2e2d77184bd

File: components/Player.tsx (hunk 1 above)
Comment (FroggyRocky) [reply components/Player.tsx]: will do

File: lib/spotify.ts (hunk 2 above)
Comment (FroggyRocky) [reply lib/spotify.ts]: will do

File: package.json (hunk 3 above)
Comment (FroggyRocky) [reply package.json]: gotcha

File: tsconfig.json (hunk 4 above)
Comment (FroggyRocky) [reply tsconfig.json]: okay, for some reason this set up + downgraded node modules allow me not to have these type errors, any suggestions ? what might be wrong with my setup ?
Screenshot 2024-11-13 at 20 26 22

Comment (FroggyRocky) [reply tsconfig.json]: Screenshot 2024-11-13 at 20 27 45

Comment (akellbl4) [reply tsconfig.json]: let me check

**Review by akellbl4 — APPROVED** [review]
@FroggyRocky I've updated all deps and adjusted tsconfig, so I can see everything is working fine. Thanks for your contribution!
